### PR TITLE
[minor_change] Optimization through an additional api call at list-identity schema endpoint to avoid retrieval of all schemas in order to get the id of a schema (DCNE-208)

### PR DESCRIPTION
**Pull request by akinross**

### CODE REVIEW

File: mso/utils.go [hunk 1]
```diff
@@ -220,3 +220,23 @@ func doPatchRequest(msoClient *client.Client, path string, payloadCon *container
 
 	return nil
 }
+
+func getSchemaIdFromName(msoClient *client.Client, name string) (string, error) {
+
+	con, err := msoClient.GetViaURL("/api/v1/schemas/list-identity")
+
+	if err != nil {
+		return "", nil
```
Comment (anvitha-jain) [inline mso/utils.go:229]: can we cahnge the return "" -> "Endpoint not found" ? 
then we can run the other part of code.

then from line 241 if return "" the we will not run the old code.

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: mso/utils.go (hunk 1 above)
Comment (akinross) [reply mso/utils.go:229]: I am not sure I follow exactly what you mean, do you want to return a string with a specific value?

Why would "" not run the old code?

The cases that I now see from the code are the following:

- errors for whatever reason -> return "", nil -> executes the else clause in line 91 of the datasource
- find the name in from response -> return schemaId, nil -> executes the else if condition in line 86 of the datasource where schemaId is a non empty string value and does a specific query for this schemaId
- does not find name in response -> return "", err -> return the error

**Review by gmicol — APPROVED** [review]
LGTM

File: mso/utils.go (hunk 1 above)
Comment (anvitha-jain) [reply mso/utils.go:229]: yes, it's clear now

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**